### DISCUSSION TITLE
how to load multi JS script(module) ?

### DISCUSSION THREAD

**Discussion by arrddon** [cables-gl/cables_docs#465] [Answers & Questions] [answered]
hello, I'm trying to using cabels and Google Firebase Storage.
To load Firebase Javascript SDK libraries, following doc, using CDN method.

![image](https://user-images.githubusercontent.com/24363682/213872229-7f7737a2-70d6-46d5-878e-85e9f8508049.png)

But it seems this method doesn't work in Cables.
**(Error Message : Cannot use import statement outside a module)**
<img width="1269" alt="image" src="https://user-images.githubusercontent.com/24363682/213872250-dacc17dd-676d-40f0-a969-6c12de663be2.png">

So I am using multiple 'CustomOP's to load multiple JS SDK, and I have to split entire JS code.
<img width="1042" alt="image" src="https://user-images.githubusercontent.com/24363682/213872794-582cb119-c67d-40a1-8325-cf4ce4d91811.png">
<img width="1055" alt="image" src="https://user-images.githubusercontent.com/24363682/213872804-57452fda-1a4c-442e-baa5-a6870c06cd60.png">

Well It works. But Sometimes an error occurs. probably because the loading order of JS SDK is different each time.

Is it possible to set the order of library loadding between 'CustomOP's?
or Can load JSs as module?

**Comment by steam0r** [comment, marked as answer]
i think the way you are doing it you are loading things multiple times. a good pattern in this case it to create one op that loads the library and creates a "parent object" that then is used by all the other ops. this way you can make sure ops only act after the library is loaded and instanciated and do not load libraries multiple times.

check ECharts or SocketCluster Ops to see that pattern in use.
